staging/src/k8s.io/api/events/v1/types.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Event is a report of an event somewhere in the cluster. It generally denotes some state change in the system.
+type Event struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// eventTime is the time when this Event was first observed. It is required.
+	EventTime metav1.MicroTime `json:"eventTime" protobuf:"bytes,2,opt,name=eventTime"`
+
+	// series is data about the Event series this event represents or nil if it's a singleton Event.
+	// +optional
+	Series *EventSeries `json:"series,omitempty" protobuf:"bytes,3,opt,name=series"`
+
+	// reportingController is the name of the controller that emitted this Event, e.g. `kubernetes.io/kubelet`.
+	// This field cannot be empty for new Events.
+	// +optional
+	ReportingController string `json:"reportingController,omitempty" protobuf:"bytes,4,opt,name=reportingController"`
+
+	// reportingInstance is the ID of the controller instance, e.g. `kubelet-xyzf`.
+	// This field cannot be empty for new Events and it can have at most 128 characters.
+	// +optional
+	ReportingInstance string `json:"reportingInstance,omitempty" protobuf:"bytes,5,opt,name=reportingInstance"`
+
+	// action is what action was taken/failed regarding to the regarding object. It is machine-readable.
+	// This field can have at most 128 characters.
+	// +optional
+	Action string `json:"action,omitempty" protobuf:"bytes,6,name=action"`
+
+	// reason is why the action was taken. It is human-readable.
+	// This field can have at most 128 characters.
+	Reason string `json:"reason,omitempty" protobuf:"bytes,7,name=reason"`
```

Comment:
Add a comments whether it is optional or required.

Reply:
Done. Also updated v1beta1.